TipCatDotNet.Api/Services/Payments/PaymentService.cs

```diff
@@ -35,39 +44,192 @@ Result Validate()
             }
 
 
-            async Task<Result<PaymentDetailsResponse>> ProceedPayment()
+            async Task<Result<PaymentIntent>> ProceedPayment()
             {
-                //TODO: use payment gateway API for proceed payment
-                return Result.Failure<PaymentDetailsResponse>($"Payment declined.");
+                var amount = Int64.Parse(paymentRequest.TipsAmount.Amount.ToString().Replace(",", ""));
```

Comment:
Use `long` for clarity

Reply:
Digit separators like `,` or `.` relates on a server locale. Better multiply the account by number of digits and throw unnecessary ones